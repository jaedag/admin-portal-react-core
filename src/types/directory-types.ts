export interface Member {
  __typename: 'Member'
  id: string
  // eslint-disable-next-line camelcase
  auth_id: string
  firstName: string
  middleName?: string
  lastName: string
  fullName: string
  idlLocation?: string
  location?: string
  nameWithTitle?: string
  currentTitle: TitleOptions
  titleConnection?: unknown
  email: string
  pictureUrl: string
  phoneNumber: string
  whatsappNumber: string
  dob: { date: string }
  maritalStatus: { status: 'Married' | 'Single' }
  gender: { gender: 'Male' | 'Female' }
  occupation: { occupation: string }
  fellowship: {
    id: string
    name: string
  }
  ministry: {
    id: string
    name: string
  }
}

export type TitleOptions = 'Pastor' | 'Reverend' | 'Bishop'
