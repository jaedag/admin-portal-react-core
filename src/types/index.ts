export * from './roles-types'
export * from './directory-types'
