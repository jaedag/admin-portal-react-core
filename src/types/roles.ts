export type Role =
  | 'leaderFellowship'
  | 'leaderBacenta'
  | 'leaderConstituency'
  | 'leaderCouncil'
  | 'leaderStream'
  | 'leaderSonta'
  | 'leaderHub'
  | 'leaderMinistry'
  | 'leaderFederalministry'
  | 'leaderGatheringService'
  | 'leaderOversight'
  | 'adminConstituency'
  | 'adminCouncil'
  | 'adminStream'
  | 'adminGatheringService'
  | 'adminOversight'
  | 'adminMinistry'
  | 'adminFederalministry'
  | 'arrivalsAdminGatheringService'
  | 'arrivalsAdminStream'
  | 'arrivalsAdminCouncil'
  | 'arrivalsAdminConstituency'
  | 'arrivalsConfirmerStream'
  | 'arrivalsCounterStream'
  | 'tellerStream'
  | 'sheepseekerStream'
  | 'all'

// CHURCHES
export type ChurchLevel =
  | 'Fellowship'
  | 'Bacenta'
  | 'Constituency'
  | 'Council'
  | 'Stream'
  | 'GatheringService'
  | 'Oversight'
  | 'Sonta'
  | 'Basonta'
  | 'Ministry'
  | 'Hub'
  | 'Federalministry'

export type ChurchLevelLower =
  | 'fellowship'
  | 'bacenta'
  | 'constituency'
  | 'council'
  | 'stream'
  | 'gatheringService'
  | 'oversight'
  | 'ministry'
  | 'hub'
  | 'sonta'
  | 'basonta'

export type CurrentUser = {
  id: string
  roles: Role[]
}
