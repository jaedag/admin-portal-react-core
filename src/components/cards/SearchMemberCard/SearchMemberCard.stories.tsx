import React from 'react'
import { Meta, StoryFn } from '@storybook/react'
import SearchMemberCard, {
  SearchMemberCardPropsInterface,
} from './SearchMemberCard'
import { Member } from '@/types'

export default {
  title: 'Components/Cards/SearchMemberCard',
  component: SearchMemberCard,
  argTypes: {
    member: { control: 'object' },
  },
} as Meta

const Template: StoryFn<SearchMemberCardPropsInterface> = (args) => (
  <SearchMemberCard {...args} />
)

export const Default = Template.bind({})
Default.args = {
  member: {
    id: '1',
    firstName: 'Dan',
    lastName: 'Abramov',
    pictureUrl: 'https://bit.ly/dan-abramov',
    fellowship: {
      id: '1',
      name: 'Tuesday Night Football',
    },
    ministry: {
      id: '1',
      name: 'Greater Love Choir',
    },
  } as Member,
  onClick: () => alert('Card clicked'),
}

export const MemberWithoutMinitry = Template.bind({})
MemberWithoutMinitry.args = {
  member: {
    id: '1',
    firstName: 'Dan',
    lastName: 'Abramov',
    pictureUrl: 'https://bit.ly/dan-abramov',
    fellowship: {
      id: '1',
      name: 'Tuesday Night Football',
    },
  } as Member,
  onClick: () => alert('Card clicked'),
}

export const MemberWithoutGroups = Template.bind({})
MemberWithoutGroups.args = {
  member: {
    id: '1',
    firstName: 'Dan',
    lastName: 'Abramov',
    pictureUrl: 'https://bit.ly/dan-abramov',
  } as Member,
  onClick: () => alert('Card clicked'),
}
