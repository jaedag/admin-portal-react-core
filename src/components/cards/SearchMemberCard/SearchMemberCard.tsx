import React from 'react'
import { Avatar, Box, Card, CardBody, Flex, Text } from '@chakra-ui/react'
import { Member } from '@/types'
import { getCloudinaryPreset } from '@/utils'

export interface SearchMemberCardPropsInterface {
  member: Member
}

const SearchMemberCard = (props: SearchMemberCardPropsInterface) => {
  const { member } = props

  return (
    <Card direction="row" marginY={2}>
      <CardBody>
        <Flex gap="5" alignContent="center" alignItems="center">
          <Avatar
            data-testid="avatar"
            name={member.firstName + ' ' + member.lastName}
            size="lg"
            src={getCloudinaryPreset(member.pictureUrl)}
          />
          <Box>
            <Text marginBottom={0} fontWeight="bold" fontSize="lg">
              {member.firstName + ' ' + member.lastName}
            </Text>

            {!!member.ministry && (
              <Text color="grey" marginTop={0}>
                {member.ministry.name}
              </Text>
            )}
          </Box>
        </Flex>
      </CardBody>
    </Card>
  )
}

export default SearchMemberCard
