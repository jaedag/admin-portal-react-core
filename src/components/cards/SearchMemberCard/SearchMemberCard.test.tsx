import React from 'react'
import { render } from '@testing-library/react'
import SearchMemberCard from './SearchMemberCard'
import { Member } from '@/types'
import '@testing-library/jest-dom'

// Mock member data
const member = {
  firstName: 'John',
  lastName: 'Doe',
  pictureUrl: 'https://example.com/avatar.jpg',
  ministry: {
    name: 'Ministry of Awesome',
  },
} as Member

describe('SearchMemberCard', () => {
  it('renders member information correctly', () => {
    const { getByText } = render(<SearchMemberCard member={member} />)

    const nameElement = getByText(`${member.firstName} ${member.lastName}`)
    expect(nameElement).toBeInTheDocument()

    const ministryElement = getByText(member.ministry?.name ?? '')
    expect(ministryElement).toBeInTheDocument()
  })

  it('matches snapshot', () => {
    const { container } = render(<SearchMemberCard member={member} />)
    expect(container).toMatchSnapshot()
  })

  it('does not render ministry if member has no ministry', () => {
    const memberWithoutMinistry = { ...member, ministry: null } as Member
    const { queryByText } = render(
      <SearchMemberCard member={memberWithoutMinistry} />
    )

    const ministryElement = queryByText(member.ministry?.name ?? '')
    expect(ministryElement).toBeNull()
  })

  it('does not render picture if member has no picture', () => {
    const memberWithoutPicture = { ...member, pictureUrl: '' } as Member
    const { queryByTestId } = render(
      <SearchMemberCard member={memberWithoutPicture} />
    )

    const pictureElement = queryByTestId('member-picture')
    expect(pictureElement).toBeNull()
  })

  it('calls onClick when clicked', () => {
    const onClick = jest.fn()
    const { getByTestId } = render(
      <SearchMemberCard member={member} onClick={onClick} />
    )

    const card = getByTestId('search-member-card')
    card.click()

    expect(onClick).toHaveBeenCalledTimes(1)
  })
})
