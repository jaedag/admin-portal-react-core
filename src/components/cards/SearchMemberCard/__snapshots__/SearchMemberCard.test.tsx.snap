// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchMemberCard matches snapshot 1`] = `
<div>
  <div
    class="chakra-card css-lgv4hl"
  >
    <div
      class="chakra-card__body css-0"
    >
      <div
        class="css-hsgtjs"
      >
        <span
          class="chakra-avatar css-1ruhx4w"
          data-testid="avatar"
        >
          <div
            aria-label="John Doe"
            class="chakra-avatar__initials css-0"
            role="img"
          >
            JD
          </div>
        </span>
        <div
          class="css-0"
        >
          <p
            class="chakra-text css-t8vhkl"
          >
            John Doe
          </p>
          <p
            class="chakra-text css-1hxe9sv"
          >
            Ministry of Awesome
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
